src/util/triplea/mapXmlCreator/CanalDefinitionsPanel.java

```diff
@@ -0,0 +1,450 @@
+    panel.layout(mapXMLCreator.getStepActionPanel());
+    mapXMLCreator.setAutoFillAction(SwingAction.of(e -> {
+      final int prevCanalCount = MapXmlHelper.getCanalDefinitionsMap().size();
+      if (prevCanalCount > 0) {
+        if (JOptionPane.YES_OPTION != MapXmlUIHelper.showYesNoOptionDialog("Auto-Fill Warning",
+            "All current canal definitions will be deleted.\rDo you want to continue with Auto-Fill?",
+            JOptionPane.WARNING_MESSAGE)) {
+          return;
+        }
+        MapXmlHelper.clearCanalDefinitions();
+      }
+      panel.clearSelection();
+
+      final Map<String, Set<String>> landWaterTerritoyConnections = MapXmlData.getLandWaterTerritoryConnections();
+
+      MapXmlHelper.validateAndAddCanalDefinitions(landWaterTerritoyConnections);
+
+      final boolean noNewCanalsBuild = MapXmlHelper.getCanalDefinitionsMap().isEmpty();
+      if (noNewCanalsBuild) {
+        JOptionPane.showMessageDialog(null, "No canals have been build!", "Auto-Fill Result",
+            JOptionPane.PLAIN_MESSAGE);
+      } else {
+        JOptionPane.showMessageDialog(null, MapXmlHelper.getHtmlStringFromCanalDefinitions(), "Auto-Fill Result",
+            JOptionPane.PLAIN_MESSAGE);
+      }
+      if (prevCanalCount > 0 || !noNewCanalsBuild) {
+        panel.repaint();
+      }
+
+    }));
+  }
+
+
+  @Override
+  protected void paintCenterSpecifics(final Graphics g, final String centerName, final FontMetrics fontMetrics,
+      final Point item, final int xTextStart) {
+    if (selectedLandTerritories.contains(centerName) || selectedWaterTerritories.contains(centerName)) {
+      final Rectangle2D stringBounds = fontMetrics.getStringBounds(centerName, g);
+      g.setColor(Color.yellow);
+      g.fillRect(Math.max(0, xTextStart - 2), Math.max(0, item.y - 6), (int) stringBounds.getWidth() + 4,
+          (int) stringBounds.getHeight());
+      g.setColor(Color.red);
+      g.drawString(centerName, Math.max(0, xTextStart), item.y + 5);
+    }
+    g.setColor(Color.red);
+  }
+
+  @Override
+  protected void paintPreparation(final Map<String, Point> centers) {}
+
+  @Override
+  protected void paintOwnSpecifics(final Graphics g, final Map<String, Point> centers) {
+    final Graphics2D g2d = (Graphics2D) g;
+    final FontMetrics fontMetrics = g.getFontMetrics();
+    for (final Entry<String, CanalTerritoriesTuple> canalDef : MapXmlHelper.getCanalDefinitionsMap()
+        .entrySet()) {
+      final Set<String> terrSet1 = canalDef.getValue().getWaterTerritories();
+      final Set<String> remainingTerrs = new TreeSet<String>(terrSet1);
+      paintOwnSpecificsToWaterTerritories(centers, g2d, fontMetrics, terrSet1, remainingTerrs, canalDef.getKey());
+      paintOwnSpecificsToLandTerritories(g, centers, canalDef);
+    }
+
+  }
+
+  public void paintOwnSpecificsToLandTerritories(final Graphics g, final Map<String, Point> centers,
+      final Entry<String, CanalTerritoriesTuple> canalDef) {
+    Set<String> terrSet1;
```

Comment:
unneccessary line here, can declare and assign the variable in one line on line 114, rather than doing it in two passes, same thing for `remainingTerrs`